fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/MetricUnionAvailability.java

```diff
@@ -148,9 +147,15 @@ public MetricUnionAvailability(@NotNull Set<PhysicalTable> physicalTables, @NotN
     }
 
     @Override
-    public SimplifiedIntervalList getAvailableIntervals(PhysicalDataSourceConstraint constraints) {
+    public SimplifiedIntervalList getAvailableIntervals(PhysicalDataSourceConstraint constraint) {
+
+        // If there are columns requested that are not configured on this table
+        if (constraint.getAllColumnPhysicalNames().size() != constraint.getAllColumnNames().size()) {
```

Comment:
This will likely always be true, due to the defaulting behavior of how mapping from `logicalToPhysicalName` works. We probably need to look at the union of the columns in the subconstraints to see if any of the requested columns are not present in the subconstraints.